encoding/series_iterator.go

```diff
@@ -111,21 +112,28 @@ func (it *seriesIterator) Close() {
 	}
 }
 
-func (it *seriesIterator) Reset(id ident.ID, nsID ident.ID, tags ident.TagIterator, startInclusive, endExclusive time.Time, replicas []Iterator) {
+func (it *seriesIterator) Replicas() []MultiReaderIterator {
+	return it.multiReaderIters
+}
+
+func (it *seriesIterator) Reset(id ident.ID, nsID ident.ID, tags ident.TagIterator, startInclusive, endExclusive time.Time, replicas []MultiReaderIterator) {
 	it.id = id
 	it.nsID = nsID
 	it.tags = tags
 	it.start = startInclusive
 	it.end = endExclusive
 	it.iters.reset()
 	it.iters.setFilter(startInclusive, endExclusive)
+	it.multiReaderIters = it.multiReaderIters[:0]
```

Comment:
Have to clear the elements before resetting the length of the slice or this leaks. i.e.

```
for idx := range it.multiReaderIters {
  it.mutliReaderIters[idx] = nil
}
it.multiReaderIters = it.multiReaderIters[:0]
```